analyzer/processes.go

```diff
@@ -0,0 +1,254 @@
+	for _, conf := range confs {
+		cmd, writer, err := command(conf)
+		if err != nil {
+			return nil, err
+		}
+		cmds = append(cmds, cmd)
+		writers = append(writers, writer)
+	}
+	writeCounter := new(writeCounter)
+	writers = append(writers, writeCounter)
+	for _, cmd := range cmds {
+		group.Go(cmd.Run)
+	}
+	group.Go(func() error {
+		_, err := io.Copy(io.MultiWriter(writers...), r)
+		for _, w := range writers {
+			if closer, ok := w.(io.Closer); ok {
+				closer.Close()
+			}
+		}
+		// Broken pipe error is a result of a process shutting down. Return nil
+		// here since the process errors are more of interest.
+		if errors.Is(err, syscall.EPIPE) {
+			err = nil
+		}
+		return err
+	})
+	return &Operation{
+		counter: writeCounter,
+		group:   group,
+	}, nil
+}
+
+func command(conf Config) (*wrappedCmd, io.WriteCloser, error) {
+	cmd := exec.Command(conf.Cmd, conf.Args...)
+	cmd.Dir = conf.WorkDir
+	pw, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+	return &wrappedCmd{
+		Cmd:         cmd,
+		stderrPath:  conf.StderrPath,
+		stderrSaver: &prefixSuffixSaver{N: 32 << 10},
+		stdoutPath:  conf.StdoutPath,
+		stdoutSaver: &prefixSuffixSaver{N: 32 << 10},
+	}, pw, nil
+}
+
+type wrappedCmd struct {
+	*exec.Cmd
+	stdoutPath  string
+	stdoutSaver *prefixSuffixSaver
+	stderrPath  string
+	stderrSaver *prefixSuffixSaver
+	wg          sync.WaitGroup
+}
+
+func (c *wrappedCmd) Run() error {
+	stderr, err := stdioWriter(c.stderrPath, c.stderrSaver)
+	if err != nil {
+		return err
+	}
+	defer stderr.Close()
+	stdout, err := stdioWriter(c.stdoutPath, c.stdoutSaver)
+	if err != nil {
+		return err
+	}
+	defer stdout.Close()
+	c.Cmd.Stderr, c.Cmd.Stdout = stderr, stdout
+	err = c.Cmd.Start()
+	if err != nil {
+		return c.error(err)
+	}
+	err = c.Cmd.Wait()
+	return c.error(err)
+}
+
+func stdioWriter(path string, saver *prefixSuffixSaver) (io.WriteCloser, error) {
+	if path == "" {
+		return zio.NopCloser(saver), nil
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		return nil, err
+	}
+	w := io.MultiWriter(f, saver)
+	return struct {
+		io.Writer
+		io.Closer
+	}{w, f}, nil
+}
+
+type nopCloser struct{}
+
+func (n nopCloser) Close() error { return nil }
+
+func (c *wrappedCmd) error(err error) error {
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		perr := &ProcessExitError{
+			Err:    exitErr,
+			Path:   c.Cmd.Path,
+			Stderr: c.stderrSaver.Bytes(),
+			Stdout: c.stdoutSaver.Bytes(),
+		}
+		return perr
+	}
+	if err != nil {
+		name := filepath.Base(c.Cmd.Path)
+		return fmt.Errorf("%s process error: %w", name, err)
+	}
+	return nil
+}
+
+type writeCounter struct {
+	written int64
+}
+
+func (w *writeCounter) Write(b []byte) (int, error) {
+	n := len(b)
+	atomic.AddInt64(&w.written, int64(n))
+	return n, nil
+}
+
+type ProcessExitError struct {
+	Err    *exec.ExitError
+	Path   string
+	Stderr []byte
+	Stdout []byte
+}
+
+func (p *ProcessExitError) Error() string {
+	builder := new(strings.Builder)
```

Comment:
```suggestion
	var builder strings.Builder
```

Reply:
Needs to be a pointer or otherwise need to pass a pointer to all times it's passed into fmt.Fprint.